src/MCPServer/lib/package.py

```diff
@@ -261,6 +264,29 @@ def _move_to_internal_shared_dir(filepath, dest, transfer):
         )
         transfer.save()
 
+    if not is_dir:
+        # Transfer is not a directory so it is an uploaded zipfile.
+        # Precreate the extraction directory for the uploaded zipfile
+        extract_dir = get_extract_dir_name(dest)
+        try:
+            os.mkdir(extract_dir)
```

Comment:
Hi @helenst is this in production at Wellcome, or are there changes upstream not in this PR?

I am working off 8ab2f81ad45ae038c43e803ac0de39c33de26b94 and I am seeing it fail with a message along the lines of:

```
Error validating BagIt package: Expected bagit.txt does not exist: /var/archivematica/sharedDirectory/currentlyProcessing/ZippedBag-718dfe84-fa1f-46b5-a172-c26f906285fa/bagit.txt
Failed bagit compliance. Not restructuring.
```

```
ZippedBag/
├── processingMCP.xml
└── ZippedBag     <--- my bag if it is called 'ZippedBag.zip'
    ├── bag-info.txt
    ├── bagit.txt
    ├── data
    │   ├── forkleaf-sundew.jpg
    │   └── roundleaf-sundew.jpg
    ├── manifest-md5.txt
    └── tagmanifest-md5.txt
```

My understanding is that this `mkdir` changes the behavior of the job entirely so that because this folder exists by the time we reach [here](https://github.com/artefactual/archivematica/blob/stable/1.10.x/src/MCPClient/lib/clientScripts/extract_bag_transfer.py#L47-L49) then 7z will extract the bag to this current directory including the bag's top level directory creating a nested structure. The behavior of the job previously was such that the directory would be created anew by the command. Hence there was never a processing config in there. 

The usual pitfall with bag transfers is that they will nearly always work if you accidentally transfer as a `standard` transfer - there's nothing special about a bag! But this doesn't seem to pass that way which is good. I wonder if there's an interaction with your zip type transfers that makes this work?

I wonder if you can take a look, and perhaps rebase this one now while there is an opportunity so it is up-to-date with the most recent MCP server code, and then I can re-review? 

I'd like to note too, there have been some large-ish changes to the server. If you take a look, and I'll look too as they're new to me, and ask around the team, I'd like to see if there are alternative solutions to this too, e.g. maybe making the server aware of the pre-configuration sooner rather than putting it into the transfer structure.

But let me know about it's use in production above, and maybe I'm not doing something right in this testing. Thanks in advance.

Reply:
I've rebased. We have been using it in production with the zip fix too and it works. It doesn't look as if the zip fix should affect this feature, but let me know how you get on with the rebased version and I'll take a closer look.

Reply:
@helenst @ross-spencer I just tested the `wellcometrust:preserve-processing-config` branch. I had to change [this Path reference](https://github.com/artefactual/archivematica/blob/7c68c90fde617bb085c97834c94f3993560f2dc0/src/MCPServer/lib/server/packages.py#L205) to be [the new LocationPath class](https://github.com/artefactual/archivematica/blob/7c68c90fde617bb085c97834c94f3993560f2dc0/src/MCPServer/lib/server/packages.py#L508) and after following Ross' steps to reproduce (creating a new processing configuration and using `httpie`) I found the same `bagit` error:

```
Error validating BagIt package: Expected bagit.txt does not exist: /var/archivematica/sharedDirectory/currentlyProcessing/ZippedBag-f12c2765-24e6-40dd-a206-7665e27a2da1/bagit.txt
Failed bagit compliance. Not restructuring.
```

Reply:
OK - it turned out the bag extraction phase needed a bit more work to ensure the processingMCP file ended up at the right level (and ensure its presence wasn't throwing everything else off)

@ross-spencer @replaceafill Let me know if it seems good to you now?